lib/basedriver/helpers.js

```diff
@@ -34,32 +35,25 @@ const APPLICATIONS_CACHE_GUARD = new AsyncLock();
 const SANITIZE_REPLACEMENT = '-';
 const DEFAULT_BASENAME = 'appium-app';
 
-
-async function cleanupCachedApps () {
+process.on('exit', () => {
   if (!APPLICATIONS_CACHE.length) {
     return;
   }
+  const appPaths = APPLICATIONS_CACHE.values()
+    .map(({fullPath}) => fullPath);
+  APPLICATIONS_CACHE.reset();
 
-  logger.debug('Performing cleanup of the applications cache');
-  const cleanupPromises = [];
-  for (const {fullPath} of APPLICATIONS_CACHE.values()) {
-    cleanupPromises.push((async () => {
-      if (await fs.exists(fullPath)) {
-        await fs.rimraf(fullPath);
-      }
-    })());
-  }
-  try {
-    await B.all(cleanupPromises);
-  } catch (e) {
-    logger.error(e);
-  } finally {
-    APPLICATIONS_CACHE.reset();
+  logger.debug(`Performing cleanup of ${appPaths.length} cached ` +
+    `application${appPaths.length > 1 ? 's' : ''}`);
+  for (const appPath of appPaths) {
+    try {
+      // Asynchronous calls are not supported in onExit handler
+      rimraf.sync(appPath);
```

Comment:
I'd rather this get wrapped inside `appium-support#fs`, where the asynchronous `rimraf` is already wrapped.

Reply:
What if we merge it as is for now and then I will commit a patch after the corresponding method is added to appium-support?

Reply:
Fine with me! Publishing `appium-support` kicks off a lot of CI work, so it's best done on off hours, anyway.